controllers/topics.controller.js

```diff
@@ -29,8 +29,15 @@ exports.getArticlesById = (req, res, next) => {
 
 exports.getCommentsByArticleId = (req, res, next) => {
   const id = req.params.article_id;
-  getCommentsDataByArticleId(id)
-    .then((comments) => {
+  const articlePromises = [
+    getCommentsDataByArticleId(id),
+    checkExists("articles", "article_id", id),
+  ];
+
+  Promise.all(articlePromises)
+    .then((results) => {
+      let comments = results[0]
+      if(!results[0] && results[1]) comments = [];
```

Comment:
I'm not sure whether this if condition is needed at all.

Reply:
Your model is set to return rows back whether it has something in it or not, so you're always going to get a results[0] as that is your comments promise that you have set to return an array either with comments in or just an empty array.  The code in this block will never run. There is also unnecessary conditional logic in the model as well.